src/core/proto/rule_table_mgr.cpp

```diff
@@ -43,6 +46,29 @@
 rule_table_mgr *g_p_rule_table_mgr = nullptr;
 static inline bool is_matching_rule(const route_rule_table_key &key, const rule_val &val);
 
+static bool is_valid_addr(struct nl_addr *addr)
```

Comment:
Duplicate code. Please make a single point of knowledge. Routing and rules are control flow, so it doesn't have to be inline.

Reply:
Thanks!